mongo/client_examples_test.go

```diff
@@ -0,0 +1,200 @@
+// Copyright (C) MongoDB, Inc. 2017-present.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License. You may obtain
+// a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+package mongo_test
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+	"go.mongodb.org/mongo-driver/mongo/readpref"
+)
+
+func ExampleConnect_ping() {
+	// Create a Client to a MongoDB server and use Ping to verify that the server is running.
+
+	clientOpts := options.Client().ApplyURI("mongodb://localhost:27017")
+	client, err := mongo.Connect(context.TODO(), clientOpts)
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer func() {
+		if err = client.Disconnect(context.TODO()); err != nil {
+			log.Fatal(err)
+		}
+	}()
+
+	// Call Ping to verify that the deployment is up and the Client was configured successfully.
+	// As mentioned in the Ping documentation, this reduces application resiliency as the server may be
+	// temporarily unavailable when Ping is called.
+	if err = client.Ping(context.TODO(), readpref.Primary()); err != nil {
+		log.Fatal(err)
+	}
+}
+
+func ExampleConnect_replicaSet() {
+	// Create and connect a Client to a replica set deployment.
+	// Given this URI, the Go driver will first communicate with localhost:27017 and use the response to discover
+	// any other nodes in the replica set.
+
+	clientOpts := options.Client().ApplyURI("mongodb://localhost:27017/?replicaSet=replset")
+	client, err := mongo.Connect(context.TODO(), clientOpts)
+	if err != nil {
+		log.Fatal(err)
+	}
+	_ = client
+}
+
+func ExampleConnect_sharded() {
+	// Create and connect a Client to a sharded deployment.
+	// The URI for a sharded cluster should specify all mongos nodes in the cluster.
```

Comment:
nodes => servers

cluster => deployment (pending decision in the other code review about which term you want to use)

Reply:
Changed to "servers" and "deployment"